sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -215,5 +233,50 @@ private ByteBufFlux bodyIntern() {
         Connection internConnection() {
             return reactorNettyConnection;
         }
+
+        private void updateSubscriptionState(Subscription subscription) {
+            if (this.state.compareAndSet(ReactorNettyResponseState.NOT_SUBSCRIBED, ReactorNettyResponseState.SUBSCRIBED)) {
+                return;
+            }
+            // https://github.com/reactor/reactor-netty/issues/503
+            // FluxReceive rejects multiple subscribers, but not after a cancel().
+            // Subsequent subscribers after cancel() will not be rejected, but will hang instead.
+            // So we need to reject ones in cancelled state.
+            if (this.state.get() == ReactorNettyResponseState.CANCELLED) {
+                throw new IllegalStateException(
+                    "The client response body has been released already due to cancellation.");
+            }
+        }
+
+        /**
+         * Called by {@link ReactorNettyClient} when a cancellation is detected
+         * but the content has not been subscribed to. If the subscription never
+         * materializes then the content will remain not drained. Or it could still
+         * materialize if the cancellation happened very early, or the response
+         * reading was delayed for some reason.
+         */
+        private void releaseAfterCancel(HttpMethod method) {
+            if (mayHaveBody(method) && this.state.compareAndSet(ReactorNettyResponseState.NOT_SUBSCRIBED, ReactorNettyResponseState.CANCELLED)) {
```

Comment:
Do we need mayHaveBody check , what happen if we clear on all cancel ?

Reply:
Since we are draining content here, we want to make sure we drain it under very specific conditions, specially when the body can be present.

Reply:
That is fine ,but my doubt is if some valid response miss mayHaveBody  (due to any missed scenario), then we will still face issue , vs draining non body too along with body response (Its a trade off thing )